wasi/wasi.go

```diff
@@ -82,9 +84,60 @@ type API interface {
 	// See https://en.wikipedia.org/wiki/Null-terminated_string
 	ArgsSizesGet(ctx wasm.HostFunctionCallContext, resultArgc, resultArgvBufSize uint32) Errno
 
-	// TODO: EnvironGet(ctx api.hostFunctionCallContext, environ, environBuf uint32) Errno
+	// EnvironGet is the WASI function that reads environment variables.
+	//
+	// There are two parameters. Both are offsets in api.HostFunctionCallContext Memory. If either are invalid due to
+	// memory constraints, this returns ErrnoFault.
+	//
+	// * environ - is the offset to begin writing environment variables offsets in uint32 little-endian encoding.
+	//   * EnvironSizesGet result environc * 4 bytes are written to this offset
+	// * environBuf - is the offset to write the null terminated environment variables in the form of "key=val", the same format as os.Environ(), to wasm.MemoryInstance.
+	//   * EnvironSizesGet result environBufSize bytes are written to this offset
+	//
+	// For example, if EnvironSizesGet wrote environc=2 and environBufSize=9 for environment variables: "a=b", "b=cd"
+	//   and EnvironGet parameters are environ=11 and environBuf=1, we expect `ctx.Memory.Buffer` to contain:
+	//
+	//                           environBufSize                 uint32le    uint32le
+	//              +------------------------------------+     +--------+  +--------+
+	//              |                                    |     |        |  |        |
+	//   []byte{?, 'a', '=', 'b', 0, 'b', '=', 'c', 'd', 0, ?, 1, 0, 0, 0, 5, 0, 0, 0, ?}
+	// environBuf --^                                          ^           ^
+	//                              environ offset for "a=b" --+           |
+	//                                         environ offset for "b=cd" --+
+	//
+	// Note: FunctionEnvironGet documentation shows this signature in the WebAssembly 1.0 (MVP) Text Format.
+	// See EnvironSizesGet
+	// See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#environ_get
+	// See https://en.wikipedia.org/wiki/Null-terminated_string
+	EnvironGet(ctx wasm.HostFunctionCallContext, resultEnviron, resultEnvironBuf uint32) Errno
```

Comment:
ack this is confusing particularly as you noted above about `ArgsGet`. in the docs, https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#environ_get doesn't classify these as results, rather params, so we should stick with that I think. that or switch `ArgsGet` params to be prefixed result. etc


However, since you have other things saying 'environ' and that's what doc says, might as well converge to that
```suggestion
	EnvironGet(ctx wasm.HostFunctionCallContext, environ, environBuf uint32) Errno
```